sudo.ps1

```diff
@@ -2,6 +2,11 @@ Set-StrictMode -Off;
 
 if(!$args) { "usage: sudo <cmd...>"; exit 1 }
 
+$powershellExe = 'powershell.exe'
```

Comment:
Why guess at this? Why not just use (get-process -id $pid).path and instantiate the same shell? This works cross platform and doesn't run the risk that you will fire up a different version pwsh than you intended (for instance if you're running powershell preview in a custom directory)

Then none of this psversiontable comparison stuff needs to be done, it'll just use whatever shell it was invoked from.

Reply:
@lukesampson I'll start a new PR if @zhoujin7 has abandoned this one, if you're interested.